apps/meteor/app/api/server/v1/im.ts

```diff
@@ -123,18 +123,35 @@ API.v1.addRoute(
 			if (!roomId) {
 				throw new Meteor.Error('error-room-param-not-provided', 'Body param "roomId" is required');
 			}
-			const canAccess = await canAccessRoomIdAsync(roomId, this.userId);
-			if (!canAccess) {
-				return API.v1.unauthorized();
+
+			let rid: IRoom['_id'] = roomId;
+			let subscription: ISubscription | null = null;
+
+			const roomExists = !!(await Rooms.findOneById(roomId));
+			if (!roomExists) {
+				// if the room doesnt exists, should check if the user has a subscription to it, and if has, close it
+				subscription = await Subscriptions.findOneByRoomIdAndUserId(roomId, this.userId);
+			} else {
+				const canAccess = await canAccessRoomIdAsync(roomId, this.userId);
+				if (!canAccess) {
+					return API.v1.unauthorized();
+				}
+
+				const { room, subscription: subs } = await findDirectMessageRoom({ roomId }, this.userId);
+
+				rid = room._id;
```

Comment:
```suggestion
				const { subscription: subs } = await findDirectMessageRoom({ roomId }, this.userId);
```
We don't really need to get the room object since we're only using its `_id` property (which we already have, `roomId`). We do know that the `roomId` provided is correct since it's been validated internally by the `findDirectMessageRoom` function.